### PR TITLE
Fix jax.scipy.stats.truncnorm cdf/sf/logcdf/logsf swallowing NaN input

### LINKED ISSUE #38606: jax.scipy.stats.truncnorm cdf/sf/logcdf/logsf swallow NaN input (return boundary 1.0/0.0 instead of NaN)
### Description

`jax.scipy.stats.truncnorm.cdf` / `sf` / `logcdf` / `logsf` silently swallow a `NaN` input, returning a support-boundary value instead of propagating `NaN`:

```python
import jax; jax.config.update('jax_enable_x64', True)
from jax.scipy.stats import truncnorm
import scipy.stats as ss
print(truncnorm.cdf(float('nan'), -1.0, 2.0))     # 1.0   (scipy: nan)
print(truncnorm.sf(float('nan'), -1.0, 2.0))      # 1.0   (scipy: nan)
print(truncnorm.logcdf(float('nan'), -1.0, 2.0))  # 0.0   (scipy: nan)
print(truncnorm.logsf(float('nan'), -1.0, 2.0))   # 0.0   (scipy: nan)
```

This is internally inconsistent with JAX's own `norm`:

```python
print(jax.scipy.stats.norm.cdf(float('nan')))     # nan  (correct)
```

The NaN appears to be clipped to the `[low, high]` support, yielding `1.0`/`0.0` instead of propagating. A NaN entering a CDF/likelihood then silently becomes a definite probability.

### Cross-check

`scipy.stats.truncnorm.{cdf,sf,logcdf,logsf}(nan, ...)` all return `nan`; `torch.distributions.Normal(0,1,validate_args=False).cdf(nan)` returns `nan` (and the validated form raises on NaN). JAX `truncnorm` alone returns finite boundary values.

### Suggested fix

Propagate NaN (e.g. a final `jnp.where(jnp.isnan(x), x, result)`), matching `norm.*` and scipy.

### System info
```
jax 0.10.1, jaxlib 0.10.1, numpy 2.4.6, python 3.12.13, CPU
```

### PR DESCRIPTION
**Pull request by Nas01010101**
Fixes #38606.

## Bug

`jax.scipy.stats.truncnorm.{cdf,sf,logcdf,logsf}` swallow a NaN `x` and return a support-boundary value instead of propagating NaN:

```python
>>> import numpy as np
>>> from scipy.stats import truncnorm as tn_np
>>> from jax.scipy.stats import truncnorm as tn_jax
>>> tn_np.cdf(np.nan, -1.0, 2.0)     # nan
>>> tn_jax.cdf(np.nan, -1.0, 2.0)    # 1.0  (wrong)
>>> tn_np.logcdf(np.nan, -1.0, 2.0)  # nan
>>> tn_jax.logcdf(np.nan, -1.0, 2.0) # 0.0  (wrong)
```

This also disagrees with `jax.scipy.stats.norm`, which already propagates NaN.

## Root cause

`logcdf` selects its result with `jnp.select`:

```python
logcdf = jnp.select(
    [x >= b, x <= a, logcdf > -0.1, x > a],
    [0, -np.inf, jnp.log1p(-jnp.exp(logsf)), logcdf],
)
```

A NaN `x` satisfies none of these conditions, so it falls through to the `jnp.select` default of `0`. `cdf` is then `exp(0) = 1`, and `sf`/`logsf` route through `logcdf` as well, so all four functions return a boundary value for a NaN input.

## Fix

Propagate NaN at the end of `logcdf` (the single chokepoint that `cdf`, `sf`, and `logsf` all route through):

```python
logcdf = jnp.where(jnp.isnan(x), np.nan, logcdf)
```

One line, and it matches scipy and `jax.scipy.stats.norm`.

## Verification

`a = -1.0`, `b = 2.0` (scipy vs JAX, before and after the fix):

| fun    | x   | scipy  | old JAX | fixed JAX |
|--------|-----|--------|---------|-----------|
| cdf    | nan | nan    | 1.0     | nan       |
| cdf    | 0.5 | 0.5497 | 0.5497  | 0.5497    |
| sf     | nan | nan    | 1.0     | nan       |
| logcdf | nan | nan    | 0.0     | nan       |
| logsf  | nan | nan    | 0.0     | nan       |

Added a regression test (`testTruncnormNanPropagation`) covering all four functions over the floating dtypes, asserting both the NaN mask and that the non-NaN values match scipy. Ran locally on CPU: `4 passed`, and confirmed the test fails when the one-line fix is reverted. Finite, `+inf`, and `-inf` inputs are unchanged and continue to match scipy.

### CODE REVIEW

**Comment by Nas01010101** [comment]
Extended the NaN propagation to the shape parameters a and b as well (since scipy returns nan for a NaN a or b too), and updated the regression test to cover NaN in x, a, and b. Verified locally against scipy.

**Comment by jakevdp** [comment]
This falls under #38651; please note that it may take a while for this to receive attention from the team.
